### DISCUSSION TITLE
Try do determine if the fcitx5 is causing Obsidian to crash on KDE Wayland when entering ";" in Chinese

### DISCUSSION THREAD

**Discussion by LumenYoung** [fcitx/fcitx5#863] [General]
As written in the title, in my case I'm experiencing this crash, I would like to know if that is the case in other machines so I can decide to whom this issue should be raised. 

The step to reproduce is:

1. Open Obsidian, locate your cursor at some position in the editor, press the ";" twice after a Chinese character. 

This is the problem in my case, it causes the obsidian to crash immediately.

The following is my system setup:

OS: Archlinux 
DE: Plasma 5.27.7
GPU: Intel Alder Lake-P
WM: KWin wayland

**Comment by wengxt** [comment]
how you get it installed? are you passing any parameter to it?

I just tried extra/obsidian in repo seems to work fine

**Reply by LumenYoung** [reply]
Thanks for the reply. Yes, I'm installing obsidian from archlinux repo. 

I found that I have several details unmentioned:

1. I use the following parameters in my desktop entry: `Exec=/usr/bin/obsidian --enable-features=UseOzonePlatform --ozone-platform=wayland --enable-wayland-ime %U`
2. I'm using ZRM for shuangpin.
3. I also tried to reproduce it without any parameters and it works fine without crashing the app.

**Reply by wengxt** [reply]
Well, I won't be surprise if chrome or electron break enable-wayland-ime again since they already did so in the past. It has nothing to do with fcitx and I can do nothing to fix their code from fcitx.

If you remove "--enable-wayland-ime" and it still crash then it's not their input method code causing crash though.

I'd suggest you just use Xwayland for it and it should just work as good as X11.

**Reply by wengxt** [reply]
I tried to look closer, seems chrome would choke on a specific input of preedit styling. Seems to be workaround-able from fcitx side

**Reply by wengxt** [reply]
should be workarounded by https://github.com/fcitx/fcitx5/commit/6a6e1693aff98085b4ee573a2119732a88e5caa0

**Comment by LumenYoung** [comment]
> should be workarounded by [6a6e169](https://github.com/fcitx/fcitx5/commit/6a6e1693aff98085b4ee573a2119732a88e5caa0)

Thanks for helping! Sorry for the late reply since I was in exam phase. 

I tested the fcitx5-git from archlinuxcn repo with the  same setup and it works for me now!

**Comment by goduck777** [comment]
I still got the same issue using the git master version of fcitx5, which include the commit 6a6e169.

Can you please help check if this issue reappear?
